citesphere-model/src/main/java/edu/asu/diging/citesphere/data/bib/impl/CitationMongoDao.java

```diff
@@ -55,18 +55,29 @@ public List<? extends ICitation> findCitations(String groupId, long start, int p
         query.limit(pageSize);
         return mongoTemplate.find(query, Citation.class);
     }
-
+    
     @Override
-    public List<? extends ICitation> findCitationsByContributorUri(List<String> groupIds, long start, int pageSize, String uri) {
-        Query query = new Query();
-        query.addCriteria(Criteria.where("group").in(groupIds));
+    public Query buildQueryToFindCitationsByContributorUri(List<String> groupIds, String uri) {
+    	Query query = new Query();
+    	query.addCriteria(Criteria.where("group").in(groupIds));
         query.addCriteria(Criteria.where("deleted").is(0));
         query.addCriteria(new Criteria().orOperator(Criteria.where("authors.uri").is(uri), Criteria.where("editors.uri").is(uri), Criteria.where("otherCreators.person.uri").is(uri)));
-        query.skip(start);
+        return query;
+    }
+
+    @Override
+    public List<? extends ICitation> findCitationsByContributorUri(List<String> groupIds, long start, int pageSize, String uri) {
```

Comment:
similar to above:
- `findCitationsByContributor`
- rename `uri` to `contributorUri`